Utils/azure-toolkit-ide-libs/azure-toolkit-ide-vm-lib/src/main/java/com/microsoft/azure/toolkit/ide/vm/VirtualMachineExplorerContributor.java

```diff
@@ -21,8 +28,13 @@
         final AzureVirtualMachine service = Azure.az(AzureVirtualMachine.class);
         return new Node<>(service).view(new AzureServiceLabelView<>(service, NAME, ICON))
                 .actions(VirtualMachineActionsContributor.SERVICE_ACTIONS)
-                .addChildren(AzureVirtualMachine::list, (vm, vmNode) -> new Node<>(vm)
+                .addChildren(this::listVirtualMachines, (vm, vmNode) -> new Node<>(vm)
                         .view(new AzureResourceLabelView<>(vm))
                         .actions(VirtualMachineActionsContributor.VM_ACTIONS));
     }
+
+    @Nonnull
+    private List<VirtualMachine> listVirtualMachines(AzureVirtualMachine s) {
+        return s.list().stream().sorted(Comparator.comparing(AbstractAzureResource::name)).collect(Collectors.toList());
```

Comment:
Do we need to extract a utils to sort azure resource?

Reply:
There is no need. We will soon migrate to `AzureResourceModule`, which will order the items in `list()`